src/sentry/uptime/endpoints/organization_uptime_stats.py

```diff
@@ -0,0 +1,175 @@
+from sentry.utils.snuba_rpc import timeseries_rpc
+
+logger = logging.getLogger(__name__)
+
+
+@region_silo_endpoint
+@extend_schema(tags=["Crons"])
+class OrganizationUptimeStatsEndpoint(OrganizationEndpoint, StatsMixin):
+    publish_status = {
+        "GET": ApiPublishStatus.EXPERIMENTAL,
+    }
+    owner = ApiOwner.CRONS
+    permission_classes = (OrganizationPermission,)
+
+    def get(self, request: Request, organization: Organization) -> Response:
+        timerange_args = self._parse_args(request)
+        projects = self.get_projects(request, organization)
+
+        uptime_subscription_ids = request.GET.getlist("uptime_subscription_id")
+
+        if not uptime_subscription_ids:
+            return self.respond("No uptime subscription ids provided", status=400)
+
+        try:
+            self._validate_uptime_subscription_ids(uptime_subscription_ids, projects)
+        except ValueError:
+            return self.respond("Invalid uptime subscription ids provided", status=400)
+
+        try:
+            responses = self._make_eap_request(
+                organization, projects, uptime_subscription_ids, timerange_args
+            )
+        except Exception:
+            logger.exception("Error making EAP RPC request for uptime check stats")
+            return self.respond("error making request", status=400)
+
+        formatted_response = self._format_response(responses)
+
+        return self.respond(formatted_response)
+
+    def _validate_uptime_subscription_ids(
+        self, uptime_subscription_ids: list[str], projects: list[Project]
+    ) -> None:
+        project_subscriptions = ProjectUptimeSubscription.objects.filter(
+            project_id__in=[project.id for project in projects],
+        ).values_list("uptime_subscription_id", flat=True)
+
+        subscriptions = UptimeSubscription.objects.filter(
+            id__in=project_subscriptions,
+            subscription_id__in=uptime_subscription_ids,
+        ).values_list("subscription_id", flat=True)
+
+        if set(uptime_subscription_ids) != set(subscriptions):
+            raise ValueError("Invalid uptime subscription ids provided")
+
+    def _make_eap_request(
+        self,
+        organization: Organization,
+        projects: list[Project],
+        uptime_subscription_ids: list[str],
+        timerange_args: StatsArgsDict,
+    ) -> list[TimeSeriesResponse]:
+        # TODO: validate that these are valid uptime subscription ids
+        start_timestamp = Timestamp()
+        start_timestamp.FromDatetime(timerange_args["start"])
+        end_timestamp = Timestamp()
+        end_timestamp.FromDatetime(timerange_args["end"])
+        request = TimeSeriesRequest(
+            meta=RequestMeta(
+                organization_id=organization.id,
+                project_ids=[project.id for project in projects],
+                trace_item_type=TraceItemType.TRACE_ITEM_TYPE_UPTIME_CHECK,
```

Comment:
lol already overloading this with a word that has nothing to do with tracing

Reply:
it is not overloading. an uptime check can be in a trace.